src/js/views/globals/search/patient-search_picklist.js

```diff
@@ -0,0 +1,261 @@
+  'itemTemplateContext',
+];
+
+const attr = 'text';
+const canClear = false;
+const isSelectList = false;
+
+const PicklistEmpty = View.extend({
+  tagName: 'li',
+  className: 'picklist--no-results',
+  template: hbs`{{ noResultsText }}`,
+  serializeData() {
+    return this.options;
+  },
+});
+
+const PicklistItem = View.extend({
+  tagName: 'li',
+  itemTemplate: hbs`{{#if icon}}{{fa icon.type icon.icon classes=icon.classes}}{{/if}}<span>{{matchText text query}}</span>`,
+  itemClassName() {
+    const classNames = [];
+
+    if (this.model.get('isDisabled')) classNames.push('is-disabled');
+    if (this.model.get('hasDivider')) classNames.push('has-divider');
+
+    return classNames.join(' ');
+  },
+  className() {
+    const classNames = ['picklist__item', 'js-picklist-item', result(this, 'itemClassName', '')];
+
+    if (this.model === this.state.get('selected')) classNames.push('is-selected');
+
+    return classNames.join(' ');
+  },
+  triggers: {
+    'click': 'select',
+  },
+  preinitialize(options) {
+    this.mergeOptions(options, ['state', ...CLASS_OPTIONS_ITEM]);
+  },
+  onRender() {
+    this.searchText = this.getItemSearchText(this.model);
+  },
+  getItemSearchText(item) {
+    return this.$el.text();
+  },
+  itemTemplateContext: noop,
+  templateContext() {
+    return extend({
+      text: this.model.get(this.attr),
+      query: this.state.get('query'),
+    }, result(this, 'itemTemplateContext'));
+  },
+  getTemplate() {
+    return this.itemTemplate;
+  },
+});
+
+const Picklist = CollectionView.extend({
+  className: 'picklist__group',
+  tagName: 'li',
+  template: hbs`
+    <div class="patient-search__picklist-header">
+      <div class="patient-search__picklist-header-name">
+        <span>Patient</span>
+      </div>
+      <div class="patient-search__picklist-header-meta">
+        <div class="patient-search__picklist-header-dob">
+          <span>DOB</span>
+        </div>
+        {{#if showIdentifierHeader}}
+          <div class="patient-search__picklist-header-identifier">
+            <span>Identifier</span>
+          </div>
+        {{/if}}
+      </div>
+    </div>
+    <ul></ul>
+  `,
+  serializeCollection: noop,
+  childViewContainer: 'ul',
+  childViewEventPrefix: 'item',
+  modelEvents: {
+    'change:query': 'filter',
+  },
+  collectionEvents: {
+    'search': 'render',
+  },
+  viewFilter(view) {
+    view.render();
+    const query = this.model.get('query');
+    return !query || !view.searchText || hasAllText(view.searchText, query);
+  },
+  initialize(options) {
+    this.mergeOptions(options, CLASS_OPTIONS_ITEM);
+  },
+  childViewOptions() {
+    const opts = pick(this, ...CLASS_OPTIONS_ITEM);
+    return extend({ state: this.model }, opts);
+  },
+  templateContext() {
+    const firstPatient = this.collection.first();
+    const identifierArray = firstPatient ? firstPatient.get('identifiers') : [];
+
+    return {
+      showIdentifierHeader: identifierArray.length > 0,
```

Comment:
Only show identifier header if the first patient result in the collection contains an identifier.

`identifers = []` vs. `identifiers = [{ type: 'PID', value: '123456789 }]`.